swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/VPNConfigurationManager.swift

```diff
@@ -21,7 +21,9 @@ enum VPNConfigurationManagerError: Error {
   }
 }
 
-public class VPNConfigurationManager {
+/// Thread-safe: Immutable wrapper around NETunnelProviderManager.
+/// Only contains a 'let' property. NETunnelProviderManager handles its own synchronisation.
+public class VPNConfigurationManager: @unchecked Sendable {
   // Persists our tunnel settings
   let manager: NETunnelProviderManager
```

Comment:
If we make `IPCClient` a stateless container (i.e. `enum`), then we could make `manager` a static var here.

Essentially, there's only ever one valid `NETunnelProviderManager` and `NETunnelProviderSession` instantiated - it might simplify the concurrency model just a tad to represent that in our abstractions.

What it seems we are doing is creating multiple instances of things that then try to access one shared thing (the `NETunnelProvider*` objects) and then doing backflips to manage access to the one shared thing. Perhaps we can reduce the abstractions around the shared thing and expose its thread safety-ness directly to callers?